.travis.yml

```diff
@@ -59,6 +63,7 @@ matrix:
       env: TARGET=go
 
 before_install:
+  - ./.travis/before-install-mvn3.3.9-trusty.sh
```

Comment:
Sorry but not sure I follow your reasoning
1) If you put the content of before-install-mvn3.3.9-trusty.sh at the beginning of before-install-linux-dotnet.sh, it will produce the desired effect for the dotnet build on ubuntu. What else do you need?
2) We don't want target-specific build actions to be global, this makes it more difficult for target owners to manage their builds, and it's up to them, not to you or me
3)4) see my reply to 1 
5) not sure you understand how travis operates. Each os/target build is run in its own fresh vm, there is no such thing as a "global" install, only installs that take place at the beginning of each particular build. In other words installing trusty globally simply damages other os/target builds (as you have experienced) with no benefit to them